File: Data Structures & Algorithms/longest-common-prefix/submission-0.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());
        string ans = "";
        string s1 = strs[0];
        string s2 = strs[strs.size()-1];
        int n = min(s1.size(), s2.size());
        for(int i=0; i<n; i++){
            if(s1[i] != s2[i])break;
            ans += s1[i];
        }
        
        return ans;
    }
};

File: Data Structures & Algorithms/combination-target-sum-ii/submission-0.java
class Solution {
    public void combination2(int idx, int target, int[]arr, List<Integer>ds, List<List<Integer>>ans){
        if(target == 0){
            ans.add(new ArrayList<>(ds));
            return;
        }
        for(int i=idx; i<arr.length; i++){
            if(i>idx && arr[i] == arr[i-1])continue;
            if(arr[i] > target){
                break;
            }
            ds.add(arr[i]);
            combination2(i+1, target-arr[i],arr, ds, ans);
            ds.remove(ds.size()-1);
        }
    }
    public List<List<Integer>> combinationSum2(int[] candidates, int target) {
        List<List<Integer>>ans = new ArrayList<>();
        Arrays.sort(candidates);
        combination2(0, target, candidates, new ArrayList<>(), ans);
        return ans;
    }
}
